erica_app/erica/request_processing/erica_input.py

```diff
@@ -78,12 +80,28 @@ class FormDataEst(BaseModel):
     stmind_gem_haushalt_count: Optional[int]
     stmind_gem_haushalt_entries: Optional[List[str]]
 
+    @validator('new_admission', always=True)
+    def if_no_tax_number_must_be_new_admission(cls, v, values):
+        if values.get('steuernummer') and v:
+            raise ValueError('can not be a new admission if tax number given')
+        if not v and not values.get('steuernummer'):
+            raise ValueError('must be new admission if no tax number given')
+        return v
+
     @validator('steuernummer')
     def must_be_correct_length(cls, v):
-        if not 10 <= len(v) <= 11:
+        if v and not 10 <= len(v) <= 11:
             raise ValueError('must be 10 or 11 numbers long')
         return v
 
+
+
+    @validator('bufa_nr', always=True)
+    def if_new_admission_bufa_nr_must_be_set_correctly(cls, v, values):
```

Comment:
Uh, that would be nice. Currently, we only validate that through Elster. However, adding this here the error would then be part of a validation error instead of a wrong bufa-number error. 

Another solution would be to check it in there the request processing.

Reply:
Ok, I don't know enough about the wiring of validations in Erica to make any meaningful suggestion, I think.

Reply:
If we raise an error during in the pydantic validator, fastapi will send its own error message and wrap the InvalidBufaNumber error inside that message.

If we would check the BufaNumber later, we could raise our own InvalidBufaNumber error and also send that to the webapp. That said, we are currently not excepting these kind of errors and if we will, we can use the validation error, too. And I think it makes more sense to define it here and ensure that only valid bufa numbers get through. Therefore I've added a check to the validator.

TLDR: I've implemented your idea 😉

Reply:
FWIW you can [override the fastapi exception handler that generates the validation error response](https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions). (Although I think we should try to work with the defaults unless we have a compelling reason not to.)